packages/client/src/index.ts

```diff
@@ -707,6 +711,96 @@ export async function connectScryptedClient(options: ScryptedClientOptions): Pro
+                    const eioEndpoint = baseUrl ? new URL(eioPath, baseUrl).pathname : '/' + eioPath;
+                    const clusterPeerOptions = {
+                        path: eioEndpoint,
+                        query: {
+                            cacehBust,
+                        },
+                        withCredentials: true,
+                        extraHeaders: {
+                            ...extraHeaders,
+                            "x-scrypted-clusterobject-port": `${port}`,
+                        },
+                        rejectUnauthorized: false,
+                        transports: options?.transports,
+                    };
+
+                    const clusterPeerSocket = new eio.Socket(explicitBaseUrl, clusterPeerOptions);
+                    let peerReady = false;
+                    clusterPeerSocket.on('close', () => {
+                        clusterPeers.delete(port);
+                        if (!peerReady) {
+                            throw new Error("peer disconnected before setup completed");
+                        }
+                    });
+
+                    try {
+                        const clusterSecretPromise = once(clusterPeerSocket, 'message');
+
+                        await once(clusterPeerSocket, 'open');
+
+                        const clusterSecret = await clusterSecretPromise as any as string;
+
+                        const serializer = createRpcDuplexSerializer({
+                            write: data => clusterPeerSocket.send(data),
+                        });
+                        clusterPeerSocket.on('message', data => serializer.onData(data));
+
+                        const clusterPeer = new RpcPeer(clientName || 'engine.io-client', "cluster-proxy", (message, reject, serializationContext) => {
+                            try {
+                                serializer.sendMessage(message, reject, serializationContext);
+                            }
+                            catch (e) {
+                                reject?.(e);
+                            }
+                        });
+                        serializer.setupRpcPeer(clusterPeer);
+                        clusterPeer.tags.localPort = connectRPCObjectPlaceholderPort;
+                        peerReady = true;
+                        return { clusterPeer, clusterSecret };
+                    }
+                    catch (e) {
+                        console.error('failure ipc connect', e);
+                        clusterPeerSocket.close();
+                        throw e;
+                    }
+                })();
+                clusterPeers.set(port, clusterPeerPromise);
+            }
+            return clusterPeerPromise;
+        };
+
+
+        const connectRPCObject = async (value: any) => {
+            const clusterObject: ClusterObject = value?.__cluster;
+            const { port, proxyId, source } = clusterObject;
```

Comment:
clusterObject may be undefined.

Reply:
the resolveObject function is missing here too, that is used to prevent connect from being called multiple times on the same object resulting in duplication.